spec.emu

```diff
@@ -0,0 +1,205 @@
+  </emu-intro>
+  <emu-clause id="sec-the-decimal-constructor">
+    <h1>The Decimal Constructor</h1>
+    <p>The constructor for the Decimal object takes a single argument. If the argument is not a String, throw a *TypeError*. The String argument must match one of the following four valid syntaxes:</p>
+    <emu-grammar name="decimal-syntax">
+      <emu-production name="PositiveDecimalDigit" type="lexical" oneof>
+	<emu-rhs>1 2 3 4 5 6 7 8 9</emu-rhs>
+      </emu-production>
+      <emu-production name="DecimalDigit" type="lexical" oneof>
+	<emu-rhs>0 | <emu-prodref name="PositiveDecimalDigit"/></emu-rhs>
+      </emu-production>
+      <emu-production name="DigitString" type="lexical">
+	<emu-rhs><emu-prodref name="DecimalDigit"/>+</emu-rhs>
+      </emu-production>
+      <emu-production name="UnsignedDecimalDigitString" type="lexical" oneof>
+	<emu-rhs><emu-prodref name="DigitString"/></emu-rhs>
+	<emu-rhs><emu-prodref name="DigitString"/> "." <emu-prodref name="DigitString"/></emu-rhs>
+      </emu-production>
+      <emu-production name="SignedDecimalDigitString" type="lexical" oneof>
+	<emu-rhs>-? <emu-prodref name="UnsignedDigitString"/></emu-rhs>
+      </emu-production>
+    </emu-grammar>
+    <emu-grammar>
+      <emu-production name="UnsignedDecimalDigitString" type="lexical" oneof>
+	<emu-rhs><emu-prodref name="DigitString"/></emu-rhs>
+	<emu-rhs><emu-prodref name="DigitString"/> "." <emu-prodref name="DigitString"/></emu-rhs>
+      </emu-production>
+      <emu-production name="SignedDecimalDigitString" type="lexical" oneof>
+	<emu-rhs>-? <emu-prodref name="UnsignedDigitString"/></emu-rhs>
+      </emu-production>
+    </emu-grammar>
+    <emu-grammar name="exponential-syntax">
+      <emu-production name="ExponentialDigitStringLHS" type="lexical" oneof>
+	<emu-rhs><emu-prodref name="PositiveDecimalDigit"/></emu-rhs>
+	<emu-rhs><emu-prodref name="PositiveDecimalDigit"/> "." <emu-prodref="DigitString"/></emu-rhs>
+      </emu-production>
+      <emu-production name="ExponentialDigitStringRHS" type="lexical">
+	<emu-rhs>-? <emu-prodref name="DigitString"/></emu-rhs>
+      </emu-production>
+      <emu-production name="SignedExponentialDigitString" type="lexical">
+	<emu-rhs>-? <emu-prodref name="ExponentialDigitStringLHS"/> ("e" | "E") <emu-prodref name="ExponentialDigitStringRHS"/></emu-rhs>
+      </emu-production>
+    </emu-grammar>
+    <emu-grammar name="infinity">
+      <emu-production>("I" | "i") ("N" | "n") | ("F" | "f") | ( ("I" | "i") | ("N" | "n") | ("I" | "i") | ("T" | "t") | ("Y" | "y") ) )?</emu-production>
+    </emu-grammar>
+    <emu-grammar name="nan">
+      <emu-production>("N" | "n") | ("A" | "a") | ("N" | "n")</emu-production>
+    </emu-grammar>
```

Comment:
I'm not sure that case-insensitive Infinity and NaN are the right thing to do here, but in any case a prose assertion would probably be clearer:
```
`Infinity` [> or any token that is an ASCII case-insensitive match]
```

Reply:
Agree. Unfortunately IEEE 754 allows any case, so this is done to be compatible with that.